Cache/Requests.lean

```diff
@@ -8,6 +8,27 @@ import Cache.Hashing
 
 namespace Cache.Requests
 
+/-- Attempts to determine the running project's GitHub repository from its `origin` Git remote. -/
+def getRemoteRepo : IO String := do
+  let out ← IO.Process.output {cmd := "git", args := #["remote", "get-url", "origin"]}
+  unless out.exitCode == 0 do
+    throw <| IO.userError s!"\
+      Failed to run Git to determine project repository (exit code: {out.exitCode}).\n\
+      Ensure Git is installed and the `origin` remote points to the project's GitHub repository.\n\
+      Stdout:\n{out.stdout.trim}\nStderr:{out.stderr.trim}\n"
+  let url := out.stdout.trim.stripSuffix ".git"
```

Comment:
What if it is an `https` clone?

Reply:
That is what the `/` (as oppose to the `:`) in the search below is designed to encapsulate. Both HTTPS and SSH GitHub URLS end in `<repo-owner>/<repo-name>` with an option a `.git` suffix.